File: WarehouseManagementWeb.Application/Dto/Output/Resource/ResourceOutput.cs
using WarehouseManagementWeb.Domain.Enums;

namespace WarehouseManagementWeb.Application.Dto.Output.Resource
{
    /// <summary>
    /// Класс выходной модели ресурса.
    /// </summary>
    public class ResourceOutput
    {
        /// <summary>
        /// Id ресурса.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Наименование ресурса.
        /// </summary>
        public string? Title { get; set; }

        /// <summary>
        /// Статус ресурса в значении перечисления.
        /// </summary>
        public DirectoryStatusEnum ResourceStatusEnum { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/directory/resource/models/output/resource-output.ts
import { DirectoryStatusEnum } from '../../../enums/directory-status-enum';

/**
 * Класс выходной модели ресурса.
 */
export class ResourceOutput {
  /**
   * Id ресурса.
   */
  id: number = 0;

  /**
   * Наименование ресурса.
   */
  title: string = '';

  /**
   * Статус ресурса в значении перечисления.
   */
  resourceStatusEnum: DirectoryStatusEnum = 0;
}
